pkg/operatorstatus/operatorstatus.go

```diff
@@ -0,0 +1,124 @@
+package operator
+
+import (
+	"fmt"
+
+	metaapi "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+
+	osapi "github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+
+	"github.com/openshift/cluster-samples-operator/pkg/apis/samplesoperator/v1alpha1"
+)
+
+func UpdateOperatorStatus(sampleResource *v1alpha1.SamplesResource) error {
+	var err error
+	if sampleResource.ConditionTrue(v1alpha1.SamplesExist) {
+		err = setOperatorStatus(sampleResource.Namespace, sampleResource.Name, osapi.OperatorAvailable, osapi.ConditionTrue, "Samples exist in the openshift project")
+	} else {
+		err = setOperatorStatus(sampleResource.Namespace, sampleResource.Name, osapi.OperatorAvailable, osapi.ConditionFalse, "Samples do not exist in the openshift project")
+	}
+	if err != nil {
+		return err
+	}
+
+	if sampleResource.ConditionTrue(v1alpha1.ConfigurationValid) {
+		err = setOperatorStatus(sampleResource.Namespace, sampleResource.Name, osapi.OperatorFailing, osapi.ConditionFalse, "The samples operator configuration is invalid")
+	} else {
+		err = setOperatorStatus(sampleResource.Namespace, sampleResource.Name, osapi.OperatorFailing, osapi.ConditionTrue, "The samples operator configuraiton is valid")
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
Should add ImportCredentials as well IMO

Now, if we do decide to add an update CVO status condition on the SamplesResource, we would forgo setting that here.

Reply:
Add it how/where?

My understanding of the samples operator flow is/was that SamplesExist can't be true until ImportCredentials is true, unless ImportCredentials is not needed(because we're not referencing registry.redhat.io)

So ImportCredentials is a secondary concern, all the CVO cares about is "has the operator done it's job" where "it's job" is "Create the samples".

In the fullness of time i'd expect this implementation to evolve to:

Check if the imagestreams have been created + imported successfully -> Available==true

but again, that's not directly dependent on looking at ImportCredentials.

Reply:
as to where, I was thinking within the SamplesExists block because
it could provide some detail in the rhel case where we are not doing the import cause the credentials are not there yet (perhaps there was a problem trying to read in the credential that could be helpful)

but if we want to keep this coarse grained that is fine ... as much of anything, getting a lay of the land here

Reply:
the status we promote up to the ClusterOperator resource is intended to be very course grained, not a duplication of the status we're putting on our own CR.  At least that's the direction i've gotten so far.

Reply:
10-4